dataprofiler/profilers/unstructured_text_profile.py

```diff
@@ -139,20 +142,20 @@ def _merge_words(self, other, merged_profile):
         :type merged_profile: TextProfiler
         :return:
         """
-        if not self._is_case_sensitive:
-            merged_profile.word_count = self.word_count.copy()
-            additive_words = other.word_count
-        else:
-            merged_profile.word_count = other.word_count.copy()
-            additive_words = self.word_count
-
-        for word in additive_words:
-            word_lower = word.lower()
-            if word_lower not in self._stop_words:
-                if self._is_case_sensitive:
-                    merged_profile.word_count[word] += additive_words[word]
-                else:
-                    merged_profile.word_count[word_lower] += additive_words[word]
+        merged_profile.word_count = self.word_count + other.word_count
+
+        # remove stop words
+        merged_profile._stop_words = self._stop_words.union(other._stop_words)
+        if len(merged_profile._stop_words) != len(self._stop_words):
+            for w, c in list(merged_profile.word_count.items()):
+                if w.lower() in merged_profile._stop_words:
+                    merged_profile.word_count.pop(w)
```

Comment:
Since `c` isn't used within the body of the loop couldn't this for loop just be `for w in merged_profile.word_count`?

Reply:
Not going to block because of this, if you update your branch I'll merge it

Reply:
I removed `c` but still keep list() as otherwise the dict will change size.